miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -48,30 +76,82 @@ public Sample get(Long sampleId) throws IOException {
+      }
     }
 
-    Sample parent = sampleDao.getSample(parentId);
+    Sample newSample = new SampleFactoryBuilder().description(sample.getDescription()).sampleType(sample.getSampleType())
+        .scientificName(sample.getScientificName()).user(user).project(sample.getProject())
+        .sampleAdditionalInfo(sample.getSampleAdditionalInfo()).sampleAnalyte(sample.getSampleAnalyte()).accession(sample.getAccession())
+        .name(sample.getName()).identificationBarcode(sample.getIdentificationBarcode()).locationBarcode(sample.getLocationBarcode())
+        .receivedDate(sample.getReceivedDate()).qcPassed(sample.getQcPassed()).alias(sample.getAlias())
+        .taxonIdentifier(sample.getTaxonIdentifier()).parent(sample.getParent()).build();
 
-    Sample newSample = new SampleFactoryBuilder().description(s.getDescription()).sampleType(s.getSampleType())
-        .scientificName(s.getScientificName()).user(user).project(project).sampleAdditionalInfo(s.getSampleAdditionalInfo())
-        .identity(s.getIdentity()).sampleAnalyte(s.getSampleAnalyte()).accession(s.getAccession()).name(s.getName())
-        .identificationBarcode(s.getIdentificationBarcode()).locationBarcode(s.getLocationBarcode()).receivedDate(s.getReceivedDate())
-        .qcPassed(s.getQcPassed()).alias(s.getAlias()).taxonIdentifier(s.getTaxonIdentifier()).parent(parent).build();
+    if (!LimsUtils.isValidRelationship(sampleValidRelationshipService.getAll(), newSample.getParent(), newSample)) {
+      throw new IllegalArgumentException("Parent " + newSample.getParent().getSampleAdditionalInfo().getSampleClass().getAlias()
+          + " not permitted to have a child of type " + newSample.getSampleAdditionalInfo().getSampleClass().getAlias());
+    }
 
-    // Check for parent and valid relationship.
-    // The relationship needs to be checked in the constructor as well.
+    try {
+      return sampleDao.addSample(newSample);
+    } catch (ConstraintViolationException e) {
+      // Send the nested root cause message to the user, since it contains the actual error.
+      throw new ConstraintViolationException(e.getMessage() + " " + ExceptionUtils.getRootCauseMessage(e), e.getSQLException(),
+          e.getConstraintName());
+    }
+  }
 
-    return sampleDao.addSample(newSample);
+  @Transactional(propagation = Propagation.MANDATORY)
+  Sample to(SampleDto sampleDto) throws IOException {
+    authorizationManager.throwIfUnauthenticated();
+    Sample sample = Dtos.to(sampleDto);
+    sample.setProject(sqlProjectDAO.get(sampleDto.getProjectId()));
+    if (sampleDto.getParentId() != null) {
+      sample.setParent(sampleDao.getSample(sampleDto.getParentId()));
+    }
+    if (sampleDto.getSampleIdentityDto() != null) {
+      sample.setIdentity(identityService.to(sampleDto.getSampleIdentityDto()));
+    }
+    if (sampleDto.getSampleAnalyte() != null) {
+      sample.setSampleAnalyte(sampleAnalyteService.to(sampleDto.getSampleAnalyte()));
```

Comment:
Do we have the structure in place to add a condition for if (sampleDto.getSampleTissue() != null) ?